pkg/apis/core/validation/validation.go

```diff
@@ -1564,7 +1564,9 @@ func validateCSIPersistentVolumeSource(csi *core.CSIPersistentVolumeSource, fldP
 		if len(csi.ControllerPublishSecretRef.Name) == 0 {
 			allErrs = append(allErrs, field.Required(fldPath.Child("controllerPublishSecretRef", "name"), ""))
 		} else {
-			allErrs = append(allErrs, ValidateDNS1123Label(csi.ControllerPublishSecretRef.Name, fldPath.Child("name"))...)
+			for _, msg := range ValidateSecretName(csi.ControllerPublishSecretRef.Name, false) {
```

Comment:
is this relaxed validation still within CSI spec?

assuming it is, and we want this change long-term, we cannot relax validation in a single release, or we'll allow persisting data the previous version of the API server considers invalid, breaking updates via that older API server. relaxing the validation has to be done over two releases, first tolerating the relaxed data in update requests of objects already containing relaxed data, then in the next release, permitting the relaxed validation on all create and update requests.

Reply:
> is this relaxed validation still within CSI spec?

Yes, CSI does not describe any Kubernetes object. A CSI driver gets content of a Secret, not its name.

Reply:
the process of relaxing validation is similar to that of permitting a new enum value in an existing field, described in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-enum-value-in-existing-field